### DISCUSSION TITLE
TypeScript type checking not performed properly?

### DISCUSSION THREAD

**Discussion by X-m7** [preactjs/wmr#944] [Q&A] [answered]
Starting from the template generated by `npm init wmr your-project-name`, renaming all the `.js` files to `.tsx` and uncommenting the `"strict": true` option in `tsconfig.json` I was able to add nonsensical statements like `const x: number = "abc";` while not getting any errors whatsoever with both the start and build commands. In fact, with prerendering disabled (which I unfortunately need to do for the project I am working on since it is a [Tauri](https://tauri.app/) app and the frontend code is useless in the Node.js context) the build command still works even with egregious mistakes like attempting to reassign a const variable or using an undeclared variable.

So, is there something I am missing here? To me the entire point of TypeScript is to be able to check things before the code runs and crashes (or otherwise misbehaves), and I am also coming from `preact-cli` (which does include proper type checking in the build command for the TypeScript template).

Also, I saw a [comment](https://github.com/preactjs/wmr/discussions/891#discussioncomment-1556152) on another discussion saying that WMR uses Sucrase to transpile TypeScript, but then looking at Sucrase's readme it states (emphasis added by me) "Compiles TypeScript code to JavaScript, removing type annotations and handling features like enums. **Does not check types.**", is that comment still accurate?

**Comment by rschristian** [comment, marked as answer]
Indeed, this behavior is correct; use `tsc` instead to check types.

Most modern build tools no longer support type checking as it's egregiously slow. It's better to leave that up to the editor or a type checking script instead.

> To me the entire point of TypeScript is to be able to check things before the code runs and crashes (or otherwise misbehaves)

This is super easy to do:

```diff
        "scripts": {
		"start": "wmr",
-		"build": "wmr build --prerender",
+		"build": "tsc && wmr build --prerender",
		"serve": "wmr serve"
	},
```
